src/Symfony/Component/Translation/Catalogue/AbstractOperation.php

```diff
@@ -147,6 +147,48 @@ public function getResult()
         return $this->result;
     }
 
+    /**
+     * @param string             $domain
+     * @param OperationInterface $operation
+     * @param string             $batch     must be one of ['all', 'obsolete', 'new']
+     */
+    public function moveMessagesToIntlDomainsIfPossible(string $batch = 'all'): void
+    {
+        if (!class_exists(\MessageFormatter::class)) {
+            return;
+        }
+
+        if (!\in_array($batch, ['all', 'obsolete', 'new'])) {
+            throw new \InvalidArgumentException('$batch argument must be one of [\'all\', \'obsolete\', \'new\'].');
+        }
+
+        foreach ($this->getDomains() as $domain) {
+            $intlDomain = $domain.MessageCatalogueInterface::INTL_DOMAIN_SUFFIX;
+            switch ($batch) {
+                case 'obsolete':
+                    $messages = $this->getObsoleteMessages($domain);
+                    break;
+                case 'new':
+                    $messages = $this->getNewMessages($domain);
+                    break;
+                case 'all':
+                default:
+                    $messages = $this->getMessages($domain);
+                    break;
+            }
+
+            if ([] === $messages || ([] === $this->source->all($intlDomain) && [] !== $this->source->all($domain))) {
```

Comment:
No need to check also the type:
`if (!$messages || (!$this->source->all($intlDomain) && $this->source->all($domain))) {`